consumption.tex

```diff
@@ -0,0 +1,50 @@
+
+The verification algorithm for $\alpha$ is presented in Algorithm~\ref{alg.verify-tx}. At the heart of the verification is the function \textsf{containsTransfer}, which takes a transaction, an address and an amount and determines if a value transfer of that exact amount to the address provided has ocured inside the transaction provided. For Bitcoin, this function locates a P2PKH output with the exact amount and receiving address. For Ethereum, the function ensures that the transaction comprises a fund transfer (i.e. the \textsf{data} field is empty), to the receiving address provided and with the exact amount. If this check passes and the transaction inclusion proofs are valid, \textsf{verify-tx} function accepts the proof $\alpha$ for the specified event $e$.
+
+\subsection{Block Connection Proof}
+
+We will look at two ways of proving block connection to the honestly adopted source chain.
+
+\subsubsection{Federation}
+A federation of $m$ nodes is picked. A federation node is responsible for running the source blockchain software and knowing at every moment the honestly adopted source blockchain. More importantly, it is responsible for transmitting this information to the target blockchain. In order to do this, we define a smart contract on the target blockchain where federation nodes are allowed to submit such information. We call this contract the \textsf{TagRepository}. A federation node encodes the honestly adopted chain into a tag and submits this tag to the \textsf{TagRepository}. Once a majority of votes is achieved, the tag is considered valid for use.
+
+Similar to FlyClient~\cite{flyclient}, we propose encoding the honestly adopted chain by using a Merkle Mountain Range (MMR). Specifically, to generate a tag we take all the chain's block ids in sequence and use them as leaves in a Merkle Mountain Range. We then use the root of that MMR as our encoding.
+
+Knowing this, it is now possible to define \textsf{blockConnection}. This component will be an MMR proof for the fact that that $id(\mathsf{blockHeader}) \in \text{MMR}$ in the form of $\mathsf{blockConnection} = [(x_0, s_0), (x_1, s_1), \dots]$, where $s_i \in \sf\{Left, Right\}$.
+
+An interesting observation is that, because the federation majority is trusted on the target blockchain, we could avoid doing the block header hashing and verification in order to verify block connection. Instead of storing the block id of each block in the MMR, we could store the transaction merkle root of each block, and again use the root of the MMR as the tag. The verifier on the smart contract side would then verify that the \textsf{txIDMerkleRoot} provided exists in the MMR using the \textsf{blockConnection}.
```

Comment:
This majority trust assumption is first articulated here. It should be properly introduced priorly.

Reply:
Here, you're leaving the reader guessing what the rest of the protocol does, and you go on to describe an alternative mechanism. You should first make sure the protocol is fully described before you do that.